nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -542,6 +542,59 @@ contract StakingEscrow is Issuer {
         }
     }
 
+    /**
+    * @notice Batch deposit. Allowed only initial deposit for each staker
+    * @param _stakers Stakers
+    * @param _numberOfSubStakes Number of sub-stakes which belong to staker in _values and _periods arrays
+    * @param _values Amount of tokens to deposit for each staker
+    * @param _periods Amount of periods during which tokens will be locked for each staker
+    */
+    function batchDeposit(
+        address[] calldata _stakers,
+        uint256[] calldata _numberOfSubStakes,
```

Comment:
Is the max number of sub-stakes for a staker 30. In which case do we need a `uint256[]` for each of these numbers? seems excessive...or is this some solidity/gas magic 🐇 🎩 ?

Reply:
I am not sure that this would do any significant optimization. In the code we have `j` / `endIndex` which can be potentially large, so if `_numberOfSubStakes` was `uint8`, we'd probably need to do a type conversion every time. Which also obstructs code readability.
And this is call parameters, not storage (where storing less could be very worthwhile)

Reply:
@michwill said everything correct. Just want to add that type reducing helps only when we talk about contract state (and then not always)

Reply:
👍